src/core/external-calendar-cache.ts

```diff
@@ -0,0 +1,286 @@
+/**
+ * External calendar caching system with expiration and LRU eviction
+ */
+
+import type { 
+  CachedCalendarData, 
+  ExternalCalendarSource,
+  CacheStats 
+} from '../types/external-calendar';
+import type { SeasonsStarsCalendar } from '../types/calendar';
+import { Logger } from './logger';
+
+interface CacheConfiguration {
+  maxSize: number;
+  defaultTtl: number; // Time to live in milliseconds
+  cleanupInterval: number;
+}
+
+interface CacheEntry {
+  data: CachedCalendarData;
+  lastAccessed: number;
+}
+
+export class ExternalCalendarCache {
+  private cache = new Map<string, CacheEntry>();
```

Comment:
This will work will for a single browser session, but we should support this surviving a reload, so we'll need to make use of `localstorage` or something similar. Ideally something that Foundry already supports/provides, maybe a custom Journal or other document type.